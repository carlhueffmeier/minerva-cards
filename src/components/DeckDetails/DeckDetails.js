import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CardBrowser from 'components/CardBrowser';
import BackButton from 'components/BackButton';
import Card from 'components/Card';

class DeckDetails extends Component {
  static propTypes = {
    history: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    deck: PropTypes.object
  };

  state = {
    selectedCard: null
  };

  handleAddNote() {
    this.props.history.push(`${this.props.location.pathname}/add-note`);
  }

  handleCardSelected(id) {
    this.setState({ selectedCard: id });
  }

  render() {
    const { deck } = this.props;

    if (!deck) {
      return (
        <div>
          <h2>Deck not found!</h2>
          <BackButton />
        </div>
      );
    }

    return (
      <div>
        <BackButton />
        <h2 className="deck-details__title">Deck: {deck.title}</h2>
        <button onClick={this.handleAddNote.bind(this)}>Add Note</button>
        <CardBrowser
          deck={deck.id}
          onSelect={this.handleCardSelected.bind(this)}
        />
        {this.state.selectedCard && <Card id={this.state.selectedCard} />}
      </div>
    );
  }
}

export default DeckDetails;
