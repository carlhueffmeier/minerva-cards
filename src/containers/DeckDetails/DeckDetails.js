import React, { Component } from 'react';
import { connect } from 'react-redux';
import { selectors as deckSelectors } from 'state/modules/decks';

export class DeckDetails extends Component {
  goBack() {
    this.props.history.goBack();
  }

  render() {
    const { title } = this.props.deck;
    return (
      <div>
        <button onClick={this.goBack.bind(this)}>Go back</button>
        <h2 className="deck-title">{title}</h2>
      </div>
    );
  }
}

function mapStateToProps(state, ownProps) {
  const { deckId } = ownProps.match.params;
  console.log(deckId);
  return {
    deck: deckSelectors.getDeckById(state, deckId)
  };
}

export default connect(mapStateToProps, null)(DeckDetails);
