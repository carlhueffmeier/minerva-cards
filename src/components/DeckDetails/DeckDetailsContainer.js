import { connect } from 'react-redux';
import { selectors as deckSelectors } from 'state/modules/decks';
import DeckDetails from './DeckDetails';

function mapStateToProps(state, ownProps) {
  const { deckId } = ownProps.match.params;
  return {
    deck: deckSelectors.getDeckById(state, deckId)
  };
}

export default connect(mapStateToProps, null)(DeckDetails);
