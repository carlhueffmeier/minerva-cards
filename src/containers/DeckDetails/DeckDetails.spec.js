import React from 'react';
import { shallow } from 'enzyme';
import { DeckDetails } from './DeckDetails';

describe('DeckDetails', () => {
  const props = {
    deck: { id: 1, title: `__TEST__` }
  };
  const deckDetails = shallow(<DeckDetails {...props} />);

  it('renders properly', () => {
    expect(deckDetails).toMatchSnapshot();
  });

  it('displays the correct title', () => {
    expect(deckDetails.find(`.deck-title`).text()).toEqual(props.deck.title);
  });
});
