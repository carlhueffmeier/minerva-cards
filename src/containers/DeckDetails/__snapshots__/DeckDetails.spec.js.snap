// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckDetails renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeckDetails
    deck={
        Object {
            "id": 1,
            "title": "__TEST__",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          Go back
</button>,
        <h2
          className="deck-title"
>
          __TEST__
</h2>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Go back",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Go back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "__TEST__",
          "className": "deck-title",
        },
        "ref": null,
        "rendered": "__TEST__",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            Go back
</button>,
          <h2
            className="deck-title"
>
            __TEST__
</h2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go back",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Go back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "__TEST__",
            "className": "deck-title",
          },
          "ref": null,
          "rendered": "__TEST__",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
